src/ImageSharp.Web/Resolvers/PhysicalFileSystemCacheResolver.cs

```diff
@@ -32,8 +33,12 @@ public PhysicalFileSystemCacheResolver(FileInfo metaFileInfo, FormatUtilities fo
         /// <inheritdoc/>
         public async Task<ImageCacheMetadata> GetMetaDataAsync()
         {
-            using Stream stream = OpenFileStream(this.metaFileInfo.FullName);
-            this.metadata = await ImageCacheMetadata.ReadAsync(stream);
+            if (this.metadata == default)
+            {
+                using Stream stream = OpenFileStream(this.metaFileInfo.FullName);
```

Comment:
This can also use the specific type instead:
```suggestion
                using FileStream stream = OpenFileStream(this.metaFileInfo.FullName);
```